src/components/state/combineStateProviders.ts

```diff
@@ -0,0 +1,10 @@
+import { StateProviders } from "./StateProviders";
+import { StateProvider } from "./StateProvider";
+
+export const combineStateProviders = 
```

Comment:
this is the thing that takes all the initial state provider functions, calls them, and reduces them onto a single global state object